test/imgix-api.js

```diff
@@ -60,25 +66,64 @@ describe('The ImgixAPI class', () => {
 });
 
 describe('ImgixAPI.prototype.request', () => {
-    var ix;
+    let ix, version;
 
     before(() => {
         ix = new ImgixAPI({
             apiKey: API_KEY
         });
+        version = ix.settings.version;
     });
 
     it('exposes a method request() on the object prototype', () => {
         assert.equal(typeof ix.request, 'function');
     });
 
     it('calling request() returns a Promise as a response', () => {
-        const req1 = ix.request("https://api.imgix.com/api/v1/assets")
-        .then(response => {
-            assert.exists(response.headers);
-        })
-        .catch(() => {});
+        const promise = ix.request(ASSETS_ENDPOINT);
+        assert.equal(typeof promise.then, 'function');
+    });
+
+    it('constructs the full API URL prior to completing request', () => {
+        const stubFetch = sinon.stub(fetchWrapper, 'fetch').returns(Promise.resolve());
+        ix.request(ASSETS_ENDPOINT)
+        .catch(resp => resp);
```

Comment:
I don't know if this should happen in this PR or not, but if we refactor the `catch` at `src/imgix-api.js:L74` to not use `throw` we shouldn't need to attach `catch()` calls in these tests.

Reply:
Yeah you're right about that. I was planning to handle that change and refactor `APIError` in a separate PR.